### PR TITLE
Add Attendees API to dredd test

### LINKED ISSUE #3924: Add Attendee API to dredd test
parent issue #3870.

### PR DESCRIPTION
**Pull request by enigmaeth**
Fixes #3924.

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

#### Changes proposed in this pull request:

- Adds Attendee API Blueprint documentation
- Adds hooks for attendee APIs
- Adds factory class for Attendee( ticket holder)

### CODE REVIEW

**Comment by enigmaeth** [comment]
@shubham-padia What about these hound violations??

File: tests/hook_main.py [hunk 1]
```diff
@@ -835,6 +836,73 @@ def ticket_delete(transaction):
...
+    :return:
+    """
+    with stash['app'].app_context():
+        attendee = AttendeeFactory()
+        db.session.add(attendee)
+        db.session.commit()
+
+
+
+@hooks.before("Attendees > Attendee Details > Delete Attendee")
```
Comment (shubham-padia) [inline tests/hook_main.py]: fix this

File: tests/hook_main.py [hunk 2]
```diff
@@ -835,6 +836,73 @@ def ticket_delete(transaction):
...
+    :return:
+    """
+    with stash['app'].app_context():
+        attendee = AttendeeFactory()
+        db.session.add(attendee)
+        db.session.commit()
+
+
+
+@hooks.before("Attendees > Attendee Details > Update Attendee")
```
Comment (shubham-padia) [inline tests/hook_main.py]: fix this

File: tests/hook_main.py [hunk 3]
```diff
@@ -835,6 +836,73 @@ def ticket_delete(transaction):
 
 
 
+# ---------------------- Attendees (Ticket Holder) ---------------------
```
Comment (shubham-padia) [inline tests/hook_main.py]: fix this

File: tests/hook_main.py [hunk 4]
```diff
@@ -835,6 +836,73 @@ def ticket_delete(transaction):
...
+
+@hooks.before("Attendees > Attendees Collection > List All Attendees")
```
Comment (shubham-padia) [inline tests/hook_main.py]: fix this everywhere in the doc i.e remove the line between `-----....` and `@hooks.befo....` everywhere

File: docs/api/api_blueprint.apib [hunk 5]
```diff
@@ -4823,6 +4823,294 @@ Delete a single ticket.
...
+| Parameter | Description | Type | Required |
+|:----------|-------------|------|----------|
+| `firstname`  | First name of the Attendee | string | **yes** |
+| `lastname` | Last name of the Attendee | string | **yes** |
+| `email` | Email of the Attendee | string | - |
+| `address` | Address of the Attendee | float | - |
+| `city` | City of the Attendee | integer | - |
+| `state` | State of the Attendee | string | - |
+| `country` | Country of the Attendee | string | - |
+| `is_ckecked_in` | If the attendee is checked in | boolean | - |
```
Comment (shubham-padia) [inline docs/api/api_blueprint.apib]: *checked

**Comment by shubham-padia** [comment]
@enigmaeth commented inline :)

File: tests/hook_main.py (hunk 1 above)
Comment (enigmaeth) [inline tests/hook_main.py]: @shubham-padia Resolved.

**Comment by niranjan94** [comment]
@enigmaeth @shubham-padia this API should be accessible as a relation to an event. `GET /v1/event/<id>/attendees` ... `GET /v1/attendees` shouldn't be allowed. 

Please open an issue and fix this ASAP. High Priority.

**Pull request merged**